File: core/src/main/java/pcb/uwu/units/finance/RentUnit.java
package pcb.uwu.units.finance;

import pcb.uwu.core.CompositeUnit;
import pcb.uwu.core.UnitCounter;
import pcb.uwu.units.base.TimeUnit;

public class RentUnit extends CompositeUnit {

	public RentUnit(MoneyUnit money, TimeUnit time) {
		super(new UnitCounter()
				.major(money)
				.minor(time));
	}

	public RentUnit(MoneyUnit money, InterestRateUnit interestRateUnit) {
		super(new UnitCounter()
				.major(money)
				.major(interestRateUnit.getUnitCounter()));
	}
}

File: core/src/main/java/pcb/uwu/unit/derived/economics/RentUnit.kt
package pcb.uwu.unit.derived.economics

import pcb.uwu.core.CompositeUnit
import pcb.uwu.core.UnitCounter
import pcb.uwu.unit.base.TimeUnit
import pcb.uwu.unit.derived.acoustics.FrequencyUnit

class RentUnit : CompositeUnit
{
    constructor(currency: CurrencyUnit,
                time: TimeUnit)
            : super(UnitCounter()
                        .major(currency)
                        .minor(time))

    constructor(currency: CurrencyUnit,
                interestRateUnit: FrequencyUnit)
            : super(UnitCounter()
                        .major(currency)
                        .major(interestRateUnit))
}